src/future/future/mod.rs

```diff
@@ -355,6 +356,18 @@ extension_trait! {
         {
             TryJoin::new(self, other)
         }
+
+        #[doc = r#"
+            Waits for both the future and a timeout, if the timeout completes before
+            the future, it returns an TimeoutError.
```

Comment:
Can you write test code following other methods?

https://docs.rs/async-std/1.2.0/async_std/future/trait.Future.html#method.race